src/Clock.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState, useEffect } from 'react';
+
+type Props = {
+  clockName: string;
+};
+
+export const Clock:React.FC<Props> = ({ clockName }) => {
+  const [currentTime, setCurrentTime] = useState(new Date());
+  const [previousClockName, setPreviousClockName] = useState(clockName);
+
+  useEffect(() => {
+    const timerId = window.setInterval(() => {
+      setCurrentTime(new Date());
+      // eslint-disable-next-line no-console
+      console.info(new Date().toUTCString().slice(-12, -4));
+    }, 1000);
+
+    return () => {
+      window.clearInterval(timerId);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (previousClockName !== clockName) {
+      // eslint-disable-next-line no-console
+      console.debug(`Renamed to ${clockName}`);
+      setPreviousClockName(clockName);
+    }
+  }, [clockName]);
+
```

Comment:
You don't need this check, this function will run if clockname from dependency array its changing, so you can remove the if statement and the cachedClockname as well.